django/db/backends/base/features.py

```diff
@@ -299,6 +299,14 @@ class BaseDatabaseFeatures:
     supports_update_conflicts = False
     supports_update_conflicts_with_target = False
 
+    # Does the backend support partial updating rows on conflicts during
+    # INSERT?
+    supports_update_conflicts_with_condition = False
+
+    # Does the backend support the special "excluded" table used to reference
+    # the values originally proposed for insertion in ON CONFLICT DO UPDATE?
+    supports_excluded_expression = False
```

Comment:
Do we need two flags or can these be the same flag?
In other words, can we just use `supports_update_conflicts_with_condition` everywhere?

Reply:
I think it's better separate as there is no guarantee that there will be this special table in other implementations. For example, if MySQL/MariaDB had supported conditional updates, we could have made the following change:

```patch
diff --git a/django/db/backends/mysql/operations.py b/django/db/backends/mysql/operations.py
index 7fabb0e05e..5bbed93a15 100644
--- a/django/db/backends/mysql/operations.py
+++ b/django/db/backends/mysql/operations.py
@@ -435,18 +435,18 @@ class DatabaseOperations(BaseDatabaseOperations):
     def on_conflict_suffix_sql(self, fields, on_conflict, update_fields, unique_fields):
         if on_conflict == OnConflict.UPDATE:
             conflict_suffix_sql = "ON DUPLICATE KEY UPDATE %(fields)s"
             # The use of VALUES() is deprecated in MySQL 8.0.20+. Instead, use
             # aliases for the new row and its columns available in MySQL
             # 8.0.19+.
             if not self.connection.mysql_is_mariadb:
                 if self.connection.mysql_version >= (8, 0, 19):
-                    conflict_suffix_sql = f"AS new {conflict_suffix_sql}"
-                    field_sql = "%(field)s = new.%(field)s"
+                    conflict_suffix_sql = f"AS EXCLUDED {conflict_suffix_sql}"
+                    field_sql = "%(field)s = EXCLUDED.%(field)s"
                 else:
                     field_sql = "%(field)s = VALUES(%(field)s)"
             # Use VALUE() on MariaDB.
             else:
                 field_sql = "%(field)s = VALUE(%(field)s)"
 
             fields = ", ".join(
                 [
```

This would have effectively allowed the excluded expression to work, but only for MySQL 8.0.19+. I realise this is hypothetical...